### ISSUE TITLE
ERROR: Unable to create print queue, ignoring printer.

### ISSUE THREAD

**Issue by paulmenzel** [OpenPrinting/cups-filters#148]
On Debian Sid/unstable, cups-browsed 1.25.4 fails to start.

```
$ systemctl status cups-browsed.service 
● cups-browsed.service - Make remote CUPS printers available locally
   Loaded: loaded (/lib/systemd/system/cups-browsed.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2019-09-03 08:35:02 CEST; 47min ago
  Process: 424 ExecStart=/usr/sbin/cups-browsed (code=exited, status=1/FAILURE)
 Main PID: 424 (code=exited, status=1/FAILURE)
```

Starting it manually with the switch `--debug`, I see the messages below.

```
$ sudo /usr/sbin/cups-browsed --debug
Tue Sep  3 09:06:54 2019 Reading command line option --debug, turning on debug mode (Log on standard error).
Tue Sep  3 09:06:54 2019 cups-browsed of cups-filters version 1.25.4 starting.
Tue Sep  3 09:06:54 2019 Reading config: BrowseRemoteProtocols dnssd cups
Tue Sep  3 09:06:54 2019 No "Browse..." line at all, accept all servers ("BrowseOrder Deny,Allow").
Tue Sep  3 09:06:54 2019 main() in THREAD 140164762878016
Tue Sep  3 09:06:54 2019 cups-browsed: Creating http connection to local CUPS daemon: /var/run/cups/cups.sock:631
Tue Sep  3 09:06:54 2019 update_netifs() in THREAD 140164762878016
Tue Sep  3 09:06:54 2019 network interface lo: Local host name/address: 127.0.0.1
Tue Sep  3 09:06:54 2019 network interface lo: Local host name/address: localhost
Tue Sep  3 09:06:54 2019 network interface enp34s0: Local host name/address: 192.168.1.2
Tue Sep  3 09:06:54 2019 network interface enp34s0: Local host name/address: XXX
Tue Sep  3 09:06:54 2019 Network interface enp34s0 at 192.168.1.2 for legacy CUPS browsing/broadcasting
Tue Sep  3 09:06:54 2019 network interface lo: Local host name/address: ::1
Tue Sep  3 09:06:54 2019 network interface enp34s0: Local host name/address: fe80::aa4b:e3c3:be18:e548
Tue Sep  3 09:06:54 2019 cups-browsed [BrowsePoll /var/run/cups/cups.sock:0]: IPP-Create-Subscription
Tue Sep  3 09:06:54 2019 cups-browsed [BrowsePoll /var/run/cups/cups.sock:0]: subscription ID=1672
Tue Sep  3 09:06:54 2019 cups-browsed (/var/run/cups/cups.sock): cupsEnumDests
Tue Sep  3 09:06:54 2019 Could not determine system default printer!
Tue Sep  3 09:06:54 2019 find_previous_queue() in THREAD 140164762878016
Tue Sep  3 09:06:54 2019 get-printer-attributes IPP call failed on printer epson_wf_2660_series (implicitclass://EPSON_WF_2660_Series/).
Tue Sep  3 09:06:54 2019 ERROR: Unable to create print queue, ignoring printer.
Tue Sep  3 09:06:54 2019 ERROR: Unable to allocate memory.
```
Without cups-browsed running, printing of course fails with the error below.

    [Job 143] No destination host name supplied by cups-browsed for printer \"EPSON_WF_2660_Series\", is cups-browsed running?

**Comment by paulmenzel** [comment]
Updating CUPS from 2.2.12-2 to 2.3.0-1 does *not* fix this.

**Comment by paulmenzel** [comment]
Downgrading the packages *cups-filters* and *cups-browsed* from 1.25.4 to 1.25.3 also does *not* help.

**Comment by paulmenzel** [comment]
No, idea if the information below helps.

    $ wget https://raw.githubusercontent.com/steveathon/cups/master/test/get-printer-attributes-2.0.test
    $ ipptool -tv ipp://EPSON73FFDE.local:631/ipp/print get-printer-attributes-2.0.test

The output is in a separate file [ipptool-tv-output.txt](https://github.com/OpenPrinting/cups-filters/files/3568319/ipptool-tv-output.txt).

**Comment by paulmenzel** [comment]
Downgrading to cups-filters and cups-browsed to 1.21.6-5 from [Debian stable/buster](https://packages.debian.org/buster/cups-browsed) does make it work again.

**Comment by tillkamppeter** [comment]
@deepak0405, could you have a look into this, it seems to be caused by your major changes.

**Comment by tillkamppeter** [comment]
@deepak0405, the printer returns correct data on the get-printer-attributes IPP request, so something seems to go wrong when you handle the data and generate the PPD file.
@paulmenzel, there was a substantial change in cups-browsed in the 1.24.0 release of cups-filters. Probably this introduced the problem.

**Comment by paulmenzel** [comment]
I submitted a bug to Debian’s Bug Tracking System, and it was assigned the number [939316](https://bugs.debian.org/939316).

**Comment by tillkamppeter** [comment]
@paulmenzel, I think submitting to Debian does not help much, as this is most probably a problem of cups-browsed itself, not of the Debian packaging.

**Comment by paulmenzel** [comment]
> @paulmenzel, I think submitting to Debian does not help much, as this is most probably a problem of cups-browsed itself, not of the Debian packaging.

In this case, this is to inform the package maintainer, and to document it for other Debian users. The Debian BTS can and should also be used for upstream issue (and I tagged it accordingly), so the opinions sometimes differ from maintainer to maintainer.

**Comment by tillkamppeter** [comment]
Fixed in commit a20ae8146. Thank you for the bug report.
Problem was that there was a left over locally generated print queue (with "implicitclass://..." URI) from a crashed (or not cleanly shut down) cups-browsed and on the next start cups-browsed picked up this queue and tried a get-printer-attributes request on the "implicitclass://..." URI and this caused a fatal error.
Now I prevented these attemts and also removed the wrong fatality.
